schema.py

```diff
@@ -117,6 +124,8 @@ def validate(self, data):
             coverage = set()  # matched schema keys
             # for each key and value find a schema entry matching them, if any
             sorted_skeys = sorted(s, key=priority)
+            # resort sorted_skeys with priority_of_dict_key
+            sorted_skeys = sorted(sorted_skeys, key=priority_of_dict_key)
```

Comment:
It would probably be clearer if the two sorting criteria were used in a single call to `sorted`. E.g. like this:

```python
sorted_skeys = sorted(sorted_skeys, key=lambda k: (priority(k), priority_of_dict_key(k)))
```